crates/task-impls/src/consensus.rs

```diff
@@ -1419,9 +1417,11 @@ where
                 .sign_validating_or_commitment_proposal::<I>(&leaf.commit());
             // TODO: DA cert is sent as part of the proposal here, we should split this out so we don't have to wait for it.
             let proposal = QuorumProposal {
-                block_commitment,
+                block_header: VIDBlockHeader {
```

Comment:
Here we should call an app-specific trait function to build the header out of the commitment and previous header (which we should have from the parent leaf, I think?)

Reply:
Sounds good. The parent leaf doesn't have the header directly, since its `deltas` field could be either the payload or the payload commitment. I'm going to add a function to build the header from the commitment and previous leaf instead.